### ISSUE TITLE
Missing Training Script Prevents Model Generation and Retraining

### ISSUE THREAD

**Issue by aaron-seq** [aaron-seq/Wood-AI-CML-ALO-ML-model#6]
## Problem Description

The repository lacks any training scripts or pipeline to generate the ML model file. While the README mentions model training with RandomForestClassifier, there is no executable script that allows users to train or retrain the model with new data. This creates a complete dependency on having a pre-trained model file and prevents any model updates or improvements.

## Affected Files

- Missing: `train.py` or `scripts/train_model.py` (training script)
- Missing: `models/` directory structure for storing trained models
- `app/main.py` (lines 27-35, expects pre-trained model)
- `README.md` (mentions training but provides no scripts)

## Technical Impact

**Severity:** High

- Cannot train initial model from scratch
- Cannot retrain model with new corrosion data
- No way to update model with improved features
- Unable to experiment with different hyperparameters
- No pipeline for model versioning
- Breaks reproducibility of reported metrics (accuracy: 0.92, f1: 0.90)

## Steps to Reproduce

1. Clone the repository
2. Search for training scripts: `find . -name "*train*"`
3. Check for model training code
4. Observe: No training scripts exist

## Expected Behavior

The repository should include:

1. Training script (`train.py`) with:
   - Data loading from CSV
   - Feature engineering pipeline
   - Model training with hyperparameters from README
   - Model serialization to `models/cml_elimination_model.joblib`
   - Performance metrics logging

2. Requirements for training:
   - scikit-learn
   - pandas
   - joblib

## Suggested Solution

Create `train.py`:

```python
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score
import joblib
import os

def train_model(data_path, model_output_path):
    # Load data
    df = pd.read_csv(data_path)
    
    # Define features from README
    features = [
        'average_corrosion_rate', 'thickness_mm', 'commodity',
        'feature_type', 'cml_shape', 'remaining_life_years',
        'corrosion_thickness_ratio', 'risk_score', 'days_since_inspection'
    ]
    
    X = df[features]
    y = df['target']
    
    # Split data
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.2, random_state=42
    )
    
    # Train model with hyperparameters from README
    model = RandomForestClassifier(
        n_estimators=100,
        max_depth=10,
        min_samples_split=5,
        random_state=42
    )
    
    model.fit(X_train, y_train)
    
    # Evaluate
    y_pred = model.predict(X_test)
    y_pred_proba = model.predict_proba(X_test)[:, 1]
    
    metrics = {
        'accuracy': accuracy_score(y_test, y_pred),
        'precision': precision_score(y_test, y_pred),
        'recall': recall_score(y_test, y_pred),
        'f1_score': f1_score(y_test, y_pred),
        'roc_auc': roc_auc_score(y_test, y_pred_proba)
    }
    
    print("Model Performance:")
    for metric, value in metrics.items():
        print(f"{metric}: {value:.3f}")
    
    # Save model
    os.makedirs(os.path.dirname(model_output_path), exist_ok=True)
    joblib.dump(model, model_output_path)
    print(f"Model saved to {model_output_path}")

if __name__ == "__main__":
    train_model(
        data_path='data/corrosion_data.csv',
        model_output_path='models/cml_elimination_model.joblib'
    )
```

Also update `README.md` to include training instructions:

```bash
# Train the model
python train.py
```
